### PR TITLE
feat: add support for transaction filtering in Ledger.balance()

### PR DESCRIPTION
**Pull request by pwnage101**
Also:
* Add unit tests for new function.
* Bump version to 0.2.1

ENT-6886

### CODE REVIEW

File: openedx_ledger/utils.py [hunk 1]
```diff
@@ -46,3 +49,26 @@ def create_idempotency_key_for_transaction(ledger, quantity, is_initial_deposit=
...
     hashed_metadata = hashlib.md5(str(idpk_data).encode()).hexdigest()
     return f'{ledger.idempotency_key}-{quantity}-{hashed_metadata}'
+
+
+def calculate_transactions_balance(input_transactions_queryset):
+    """
+    Calculate the balance of the given transactions, taking reversals into account.
+
+    WARNING: All transactions should be part of the same ledger, or at least the same enterprise customer, or else the
+    output probably wouldn't be useful for anything.
```
Comment (iloveagent57) [inline openedx_ledger/utils.py:59]: small, non-blocking idea: you could enforce this via the method signature by requiring a `ledger` argument and then tacking a `.filter(ledger=ledger)` onto your `transactions` expression below.

Comment (iloveagent57) [reply openedx_ledger/utils.py:59]: or alternatively, make this a method of the `Ledger` model and add a `.filter(ledger=self)` to the queryset provided as input.

**Review by iloveagent57 — APPROVED** [review]

File: openedx_ledger/utils.py (hunk 1 above)
Comment (pwnage101) [reply openedx_ledger/utils.py:59]: yea, that sounds like a good idea to keep it part of the ledger model.

Comment (pwnage101) [reply openedx_ledger/utils.py]: Updated to add just a `transactions_filter` dict parameter to the Ledger.balance() function.

**Review by iloveagent57 — APPROVED** [review]

**Pull request merged**